assets/src/edit-story/app/font/fontProvider.js

```diff
@@ -53,13 +71,33 @@ function FontProvider({ children }) {
 
   const maybeEnqueueFontStyle = useLoadFontFiles();
 
+  const getMenuFonts = useCallback((fontFamilyList) => {
+    const newFontList = fontFamilyList.filter(
+      (fontName) => loadedFontFamily.current.indexOf(fontName) < 0
+    );
+    if (!newFontList?.length) {
+      return new Promise((resolve) => resolve(''));
+    }
+    return fetch(
+      `${GOOGLE_MENU_FONT_URL}?family=${encodeURIComponent(
+        newFontList.join('|')
+      )}&subset=menu`
+    ).then((response) => {
+      loadedFontFamily.current = [...loadedFontFamily.current, ...newFontList];
```

Comment:
This is a race condition. If a request with `A,B,C` fonts and then shortly after another request comes with `B,C,D`, then `B` and `C` will be loaded twice. I looked in the stylesheet and there are a lot of duplicates there. Essentially any scroll operation leads to lots of dupes. You can in theory do the `loaded = [...loaded, ...new]` before you proceed to the fetch, that would remove the race condition. But the risk is that you will remember failures. To address that, you could always do `fetch(...).catch(() => /*remove newFonts from the loaded list*/)`.

This might explain some other weirdness I'm seeing as well. But at this point most of font previews are not showing up for me.

Reply:
Done